test/com/google/enterprise/adaptor/DocumentHandlerTest.java

```diff
@@ -402,6 +406,56 @@ public void getDocContent(Request request, Response response)
   }
 
   @Test
+  public void testContentTransform() throws Exception {
+    ContentTransformFactory contentTransformFactory =
+        new ContentTransformFactory(
+        new ArrayList<Map<String, String>>() {
+          {
+            add(new HashMap<String, String>() {
+              {
+                put("class", SampleDocumentContentTransform.class.getName());
+              }
+            });
+          }
+        });
+    mockAdaptor = new MockAdaptor() {
+      @Override
+      public void getDocContent(final Request request, final Response response)
+          throws IOException, InterruptedException {
+        response.setContentType("image/jpeg");
+        OutputStream os = response.getOutputStream();
+        os.write("some cool stuff".getBytes(Charsets.UTF_8));
+        os.close();
+      }
+    };
+    String remoteIp = ex.getRemoteAddress().getAddress().getHostAddress();
+    DocumentHandler handler = createHandlerBuilder()
+        .setAdaptor(mockAdaptor)
+        .setFullAccessHosts(new String[]{remoteIp})
+        .setContentTransformPipeline(contentTransformFactory)
+        .build();
+    handler.handle(ex);
+    assertEquals("some changed stuff", new String(ex.getResponseBytes()));
+  }
+
+  private static class SampleDocumentContentTransform
+      extends DocumentContentTransform {
+
+    public SampleDocumentContentTransform(Map<String, String> config) {
+      super(config);
+    }
+
+    @Override
+    public void write(final byte[] b) throws IOException {
+      if (contentType.equals("image/jpeg")) {
```

Comment:
can the content transform change the contet-type string?

That is say we put in a PDF and content transform makes it a text file.  Can we have the content transform put back the content-type string in DocumentHandler?

Let's not do that in this CL, but let's think about whether it is doable, and leave a TODO. One thing that may be a challange is that we need to know the resulting content type before we send a single byte of the response body (because the content-type ahs to be in HTTP header).

I think it will be fine, i think we will be able to have the content transform do something like:

String getContentTypeOutputForContentTypeInput(String ctIn) {
  if ("text/PDF".equalsIgnoreCase(ctIn)) {
    return "text/plain"
  }
  return ctIn;
}